### PR TITLE
Deprecating IntentMetadata.displayname

### LINKED ISSUE #312: The `AppIntent` interface incorrectly describes an intent
## Minor Issue

The `AppIntent` interface incorrectly describes an intent. The `displayName` of the `AppIntent`'s `intent` is ambiguous.

### Area of Issue

[ ] App Directory
[X] API
[ ] Context Data
[X] Intents
[ ] Use Cases
[ ] Other

### Issue Description:

Let's say we have the following FDC3 application definitions (this is just an example to illustrate the issue - some required properties are missing):

```json
[
    {
        "name": "ChartA",
        "intents": [
            {
                "name": "fdc3.ViewChart",
                "displayName": "Chart A's View Chart Intent",
                "contexts": [
                    "fdc3.instrument"
                ]
            }
        ]
    },
    {
        "name": "ChartB",
        "intents": [
            {
                "name": "fdc3.ViewChart",
                "displayName": "Chart B's View Chart Intent",
                "contexts": [
                    "fdc3.instrument"
                ]
            }
        ]
    }
]
```

After calling `fdc3.findIntent('fdc3.ViewChart')` we would get the following `AppIntent`:

```js
{
    intent: {
        name: 'fdc3.ViewChart',
        displayName: '...'
    },
    apps: [
        {
            name: 'ChartA'
        },
        {
            name: 'ChartB'
        }
    ]
}
```

The issue is that the display name is ambiguous - it could be `Chart A's View Chart Intent` or `Chart B's View Chart Intent`.

Going through the existing open-source FDC3 implementations we noticed that both [Nick Kolba's desktop-agent](https://github.com/finos/fdc3-desktop-agent/blob/eb2021491323c12713b60a621fcfd89c2ef52bf7/src/bg-listeners.ts#L835), as well as [Finsemble's FDC3 implementation](https://github.com/ChartIQ/finsemble-fdc3/blob/f69078551ea5b8e6dcc41744bbd3ec09eb3f8a3c/src/services/FDC3/desktopAgent.ts#L99) would return an `AppIntent` with the intent display name of the first application.

### Additional Context:

A PR for this issue would require a change to the existing `AppIntent` interface.

### PR DESCRIPTION
**Pull request by kriswest**
resolves #312

Marking IntentMetadata.displayName as deprecated as it can only come from app definitions in appD, which means the DA needs to pick one and all could be different. SWG agreed to go with the intent name instead as thats always the same AND is required to be recognizable

### CODE REVIEW

File: website/static/schemas/next/app-directory.yaml [hunk 1]
```diff
@@ -607,7 +607,8 @@ components:
       properties:
         displayName:
           type: string
-          description: An optional display name for the intent that may be used in UI instead of the name.
+          description: Optional display name for the intent. Deprecated in favour of the Intent name, which is common amongst all apps that support it, where teh display name may vary.
```
Comment (greyseer256) [inline website/static/schemas/next/app-directory.yaml:610]:
```suggestion
          description: Optional display name for the intent. Deprecated in favour of the Intent name, which is common amongst all apps that support it, where the display name may vary.
```

File: src/app-directory/specification/appd.yaml [hunk 2]
```diff
@@ -607,7 +607,8 @@ components:
       properties:
         displayName:
           type: string
-          description: An optional display name for the intent that may be used in UI instead of the name.
+          description: Optional display name for the intent. Deprecated in favour of the Intent name, which is common amongst all apps that support it, where the display name may vary.
```
Comment (greyseer256) [inline src/app-directory/specification/appd.yaml:610]: Should these usages of `intent` have the same capitalization? Is the capitalized one referencing the Intent concept? Sorry I'm still kind of fuzzy on when capitalization is appropriate for some of these words.

File: docs/api/ref/Metadata.md [hunk 3]
```diff
@@ -302,9 +302,10 @@ interface IntentMetadata {
   /** The unique name of the intent that can be invoked by the raiseIntent call. */
   readonly name: string;
 
-  /** A friendly display name for the intent that should be used to render UI
-   *  elements.
-   */
+  /** Display name for the intent. Deprecated in favour of using the intent name for display as
+   *  display name may vary for each application as it is defined in the app's AppD record. 
+   *  @deprecated
+  */
```
Comment (mattjamieson) [inline docs/api/ref/Metadata.md:308]:
```suggestion
  /** Display name for the intent.
   * @deprecated Use the intent name for display as display name may vary for
   * each application as it is defined in the app's AppD record.
   */
```

File: src/app-directory/specification/appd.yaml (hunk 2 above)
Comment (mattjamieson) [reply src/app-directory/specification/appd.yaml:610]: I think it should be lowercase...

File: website/static/schemas/next/app-directory.yaml (hunk 1 above)
Comment (mattjamieson) [inline website/static/schemas/next/app-directory.yaml:610]:
```suggestion
          description: Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary.
```

File: src/app-directory/specification/appd.yaml (hunk 2 above)
Comment (kriswest) [reply src/app-directory/specification/appd.yaml:610]: Lets go with lowercase - its that way in the glossary and I think appropriate in all cases other than when we are specifically discussing the concept of Intents.

 ```suggestion
          description: Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary as it is defined in the app's AppD record.
```

File: website/static/schemas/next/app-directory.yaml (hunk 1 above)
Comment (kriswest) [reply website/static/schemas/next/app-directory.yaml:610]:
```suggestion
          description: Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary as it is defined in the app's AppD record.
```

File: src/app-directory/specification/appd.yaml [hunk 4]
```diff
@@ -607,7 +607,8 @@ components:
       properties:
         displayName:
           type: string
-          description: An optional display name for the intent that may be used in UI instead of the name.
+         description: Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary as it is defined in the app's AppD record.
```
Comment (kriswest) [inline src/app-directory/specification/appd.yaml:610]:
```suggestion
          description: Optional display name for the intent. Deprecated in favour of the intent name, which is common amongst all apps that support it, where the display name may vary as it is defined in the app's AppD record.
```

**Comment by kriswest** [comment]
Changes applied - needs signing off again @mattjamieson @greyseer256

**Comment by kriswest** [comment]
@greyseer256 this'll need your review before its merged

**Review by greyseer256 — APPROVED** [review]
👍

**Pull request merged**